src/Pretzel.Logic/Templating/JekyllEngineBase.cs

```diff
@@ -113,42 +113,147 @@ private void ProcessFile(string outputDirectory, Page page, Page previous, Page
                 page.OutputFile = page.OutputFile.Replace(extension, ".html");
             }
 
-            var pageContext = PageContext.FromPage(Context, page, outputDirectory, page.OutputFile);
+            var listType = page.Bag.ContainsKey("list") ? page.Bag["list"] as string : null;
```

Comment:
The idea is good, but it will be better if this can be made more generic in order to handle custom lists too, like a serie of posts on the same subject.
And is it possible to factorize the code?